### PR TITLE
Replace gym.wrappers.FlattenDictWrapper

### PR DESCRIPTION
**Pull request by ManifoldFR**
## Description

I replaced the use of a deprecated (and removed wrapper) from Gym.

## Motivation and Context

This is deprecated in recent versions of Gym, as illustrated in issue openai/gym#1051, and replaced by `FlattenObservation` which doesn't need to have keys supplied. I used a try/except to ensure people using old builds of Gym don't get an error.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:
- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by ManifoldFR** [comment]
For the changelog, I'd have to make a new section for the WIP release?

**Comment by araffin** [comment]
>For the changelog, I'd have to make a new section for the WIP release?

yes, please

File: utils/utils.py [hunk 1]
```diff
@@ -23,9 +23,11 @@
 
 def flatten_dict_observations(env):
     assert isinstance(env.observation_space, gym.spaces.Dict)
-    keys = env.observation_space.spaces.keys()
-    return gym.wrappers.FlattenDictWrapper(env, dict_keys=list(keys))
-
+    try:
+        return gym.wrappers.FlattenObservation(env)
+    except:
```
Comment (araffin) [inline utils/utils.py:28]: please do not use bare except

Comment (ManifoldFR) [reply utils/utils.py:28]: Will fix, thanks for the feedback

**Review by araffin — APPROVED** [review]
LGTM =)

**Pull request merged**